packages/amplication-client/src/VersionControl/CommitsPage.tsx

```diff
@@ -20,42 +27,81 @@ type Props = AppRouteProps & {
   }>;
 };
 
+const MAX_ITEMS_PER_LOADING = 20;
+
 const CommitsPage: React.FC<Props> = ({ match, moduleClass }) => {
   const commitId = match.params.commit;
   const history = useHistory();
 
   const { currentProject, currentWorkspace } = useContext(AppContext);
+  const [commitsCount, setCommitsCount] = useState(1);
+  const [disableLoadMore, setDisableLoadMore] = useState(false);
+
+  const { commits, commitsError, commitsLoading, refetchCommits } = useCommit(
+    MAX_ITEMS_PER_LOADING
+  );
+
+  const [commitsData, setCommitsData] = useState<Commit[]>([]);
+
+  useEffect(() => {
+    if (commits?.length < 0 || commitsCount > 1) return;
+    setCommitsData(commits);
+  }, [commits]);
+
+  useEffect(() => {
+    setCommitsCount(1);
+  }, []);
+
+  const handleOnLoadMoreClick = useCallback(() => {
+    setCommitsCount(commitsCount + 1);
+    const getNextCommits = {
+      variables: {
+        skip: commitsCount * MAX_ITEMS_PER_LOADING,
+        take: MAX_ITEMS_PER_LOADING,
+      },
+    };
 
-  const { commits, commitsError, commitsLoading } = useCommit();
+    refetchCommits(getNextCommits.variables).then((results) => {
```

Comment:
you should only expose refetch from hook and invoke it. the hook will manage the logic